### PR TITLE
correctly access javabase in java_integration_test

### PR DESCRIPTION
**Pull request by buchgr**
java_integration_test assumes that $(JAVABASE) is always a runfiles
relative path. However, that's not correct as it purely depends on
the usage of java_runtime.

For example,

java_runtime(
name = "abs_path_jdk",
java_home = "/usr/local/lib/jdk",
)

when running with --javabase=//:abs_path_jdk then all $(JAVABASE)
make variables will contain "/usr/local/lib/jdk".

Progress towards #8033

### CODE REVIEW

File: src/test/shell/integration/java_integration_test.sh [hunk 1]
```diff
@@ -15,24 +15,52 @@
...
+javabase=${javabase%/bin/java}
+
 add_to_bazelrc "build --package_path=%workspace%"
 
 #### HELPER FUNCTIONS ##################################################
 
 function setup_local_jdk() {
   local -r dest="$1"
-  local -r src="${BAZEL_RUNFILES}/${runfiles_relative_javabase}"
+  local -r src="${java_home}"
```
Comment (laszlocsomor) [inline src/test/shell/integration/java_integration_test.sh]: Where is `java_home` defined?

Comment (buchgr) [reply src/test/shell/integration/java_integration_test.sh]: done.

File: src/test/shell/integration/java_integration_test.sh [hunk 2]
```diff
@@ -15,24 +15,52 @@
 # limitations under the License.
 #
 # These are end to end tests for building Java.
-CURRENT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
-source "${CURRENT_DIR}/../shell_utils.sh" \
+# --- begin runfiles.bash initialization ---
```
Comment (laszlocsomor) [inline src/test/shell/integration/java_integration_test.sh]: You didn't copy `set -euo pipefail`, please do so.

**Comment by jin** [comment]
What's the current status of this PR?

**Comment by laszlocsomor** [comment]
The PR is not yet merged.

**Comment by ulfjack** [comment]
This fails in CI. All of them.

"line 48: CURRENT_DIR: unbound variable"

**Pull request closed without merging**